src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,206 @@
+
+            if (!IsSingleStatementOrExpressionMethod((TMethodDeclarationSyntax)calleeMethodDeclarationNode))
+            {
+                return;
+            }
+
+            var statementContainsCallee = GetStatementContainsCallee(calleeMethodInvocationNode);
+            if (statementContainsCallee == null)
+            {
+                return;
+            }
+
+            var invocationOperation = semanticModel.GetOperation(calleeMethodInvocationNode, cancellationToken);
+            if (!(invocationOperation is IInvocationOperation))
+            {
+                return;
+            }
+
+            var codeAction = new CodeAction.DocumentChangeAction(
+                string.Format(FeaturesResources.Inline_0, calleeMethodSymbol.ToNameDisplayString()),
+                cancellationToken => InlineMethodAsync(
+                    document,
+                    calleeMethodInvocationNode,
+                    (IMethodSymbol)calleeMethodSymbol,
+                    (TMethodDeclarationSyntax)calleeMethodDeclarationNode,
+                    statementContainsCallee,
+                    (IInvocationOperation)invocationOperation,
+                    cancellationToken));
+
+            context.RegisterRefactoring(codeAction);
+        }
+
+        private async Task<Document> InlineMethodAsync(
+            Document document,
+            SyntaxNode calleeMethodInvocationSyntaxNode,
+            IMethodSymbol calleeMethodSymbol,
+            TMethodDeclarationSyntax calleeMethodDeclarationSyntaxNode,
+            SyntaxNode statementContainsCallee,
+            IInvocationOperation invocationOperation,
+            CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodParametersInfo = MethodParametersInfo.GetMethodParametersInfo(_syntaxFacts, invocationOperation);
+            var inlineContext = await InlineMethodContext.GetInlineContextAsync(
+                this,
+                document,
+                calleeMethodInvocationSyntaxNode,
+                calleeMethodSymbol,
+                calleeMethodDeclarationSyntaxNode,
+                statementContainsCallee,
+                methodParametersInfo,
+                cancellationToken).ConfigureAwait(false);
+
+            var documentEditor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var statementContainsCalleeInvocationExpression = inlineContext.StatementContainingCallee;
+            foreach (var statement in inlineContext.StatementsToInsertBeforeCallee)
+            {
+                documentEditor.InsertBefore(
+                    statementContainsCalleeInvocationExpression,
+                    // Make sure the statement is aligned with the existing statement
+                    statement.WithLeadingTrivia(statementContainsCalleeInvocationExpression.GetLeadingTrivia()));
```

Comment:
if the statementContainsCalleeInvocationExpression has a leading comment, this will duplicate to all other comments.  fix+add test

Reply:
:) Is there any good way to find the correct indentation in a method?